controller/internal/enforcer/applicationproxy/tcp/tcp.go

```diff
@@ -330,7 +335,17 @@ func (p *Proxy) StartClientAuthStateMachine(downIP net.IP, downPort int, downCon
 			if err := downConn.SetWriteDeadline(time.Now().Add(5 * time.Second)); err != nil {
 				return false, err
 			}
-			token, err := p.tokenaccessor.CreateSynPacketToken(puContext, &conn.Auth)
+
+			// We now generate the claims header
+			compactPKI, ok := p.secrets.(*secrets.CompactPKI)
+			if !ok {
+				zap.L().Error("Secrets does not hold compactPKI type, so type assertion failed", zap.Reflect("type", reflect.TypeOf(p.secrets)))
+				return isEncrypted, secrets.ErrNotCompactPKI
+			}
+			claimsHeaderBytes := claimsheader.NewClaimsHeader(
+				claimsheader.OptionCompressionType(compactPKI.Compressed),
+			).ToBytes()
+			token, err := p.tokenaccessor.CreateSynPacketToken(puContext, &conn.Auth, claimsHeaderBytes)
 			if err != nil {
```

Comment:
This should be done once ahead of time and not on every request.

Reply:
> Needs refactoring and not cut-and-paste of the same section of code.

The claimsheader is not part of datapath, it is created to share information between two endpoints (for ex: appSYN to netSYN). To be more precise, I can't put create this claimsheader outside this function, 
```
	claimsHeaderBytes := claimsheader.NewClaimsHeader(
		claimsheader.OptionCompressionType(compactPKI.Compressed),
		claimsheader.OptionDatapathVersion(d.datapathVersion),
		claimsheader.OptionEncrypt(conn.PacketFlowPolicy.Action.Encrypted()),
	).ToBytes()
```

The third argument passed is the encryption state at that particular point. I can remove the other fields from the claimsheader but @satyamsi wants to have them in all headers.